src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -191,53 +202,61 @@ public IHostBuilder CreateHostBuilder(IConsole console, string[] urls, string[]
                             urls = urls.Concat(metricUrls).ToArray();
                         }
 
-                        bool boundListeningPort = false;
-
                         //Workaround for lack of default certificate. See https://github.com/dotnet/aspnetcore/issues/28120
                         options.Configure(context.Configuration.GetSection("Kestrel")).Load();
 
+                        bindingResults.AnyBoundPorts = false;
+
                         //By default, we bind to https for sensitive data (such as dumps and traces) and bind http for
                         //non-sensitive data such as metrics. We may be missing a certificate for https binding. We want to continue with the
                         //http binding in that scenario.
-                        foreach (BindingAddress url in urls.Select(BindingAddress.Parse))
+                        foreach (string url in urls)
                         {
+                            BindingAddress address = null;
+                            try
+                            {
+                                address = BindingAddress.Parse(url);
+                            }
+                            catch (Exception ex)
```

Comment:
Is there a more targeted exception we can catch here?

Reply:
I could, but I think this is appropriate since we probably don't want to crash due to a badly named address. All exceptions are caught for just parsing the address and are logged in Startup.Configure